test/unit/org/apache/cassandra/cql3/ViewTest.java

```diff
@@ -1531,4 +1536,119 @@ private static int runningViewBuilds()
                       .map(p -> p.activeTasks.getValue() + p.pendingTasks.getValue())
                       .orElse(0);
     }
+
+    @Test
+    public void testFunctionInWhereClause() throws Throwable
```

Comment:
This test tests the fix, but it doesn't really expose the problem that we're fixing.
Unfortunately, the reproduction scenario that I found requires node restart. The original scenario in the ticket doesn't work (or silently assumes a restart). Because of that I settled on using the original test.

Reply:
if restarting is involved maybe you would like to implement a jvm dtest - you can find some examples at: `test/distributed/org/apache/cassandra/distributed/test` - and what is great, you can run them directly from IntelliJ

Reply:
IIUC ensuring the function is correctly quoted upon 'serialization' of the MV is enough and the test does indeed do check that. The restart was just one way to repro the error. My 2cts.